pandas/core/internals/concat.py

```diff
@@ -183,8 +183,9 @@ def get_reindexed_values(self, empty_dtype, upcasted_na):
                         is_datetime64tz_dtype(empty_dtype)):
                     if self.block is None:
                         array = empty_dtype.construct_array_type()
-                        missing_arr = array([fill_value], dtype=empty_dtype)
-                        return missing_arr.repeat(self.shape[1])
+                        # Workaround no DatetimeArray.repeat
```

Comment:
I'd be happy to implement this if it doesn't conflict with the requested datetimelike PR freeze.  Easy cut/paste from DatetimeIndexOpsMixin